src/metabase/util/performance.clj

```diff
@@ -131,3 +131,34 @@
   "Like `clojure.core/some` but uses our custom `reduce` which in turn uses iterators."
   [f coll]
   (unreduced (reduce #(when-let [match (f %2)] (reduced match)) nil coll)))
+
+(defn concat
+  "Like `clojure.core/concat` but accumulates the result into a vector."
+  ([a b]
+   (into (vec a) b))
+  ([a b c]
+   (as-> (transient (vec a)) res
+     (reduce conj! res b)
+     (reduce conj! res c)
+     (persistent! res)))
+  ([a b c d]
+   (as-> (transient (vec a)) res
+     (reduce conj! res b)
+     (reduce conj! res c)
+     (reduce conj! res d)
+     (persistent! res)))
+  ([a b c d e]
+   (as-> (transient (vec a)) res
+     (reduce conj! res b)
+     (reduce conj! res c)
+     (reduce conj! res d)
+     (reduce conj! res e)
+     (persistent! res)))
+  ([a b c d e f]
+   (as-> (transient (vec a)) res
+     (reduce conj! res b)
+     (reduce conj! res c)
+     (reduce conj! res d)
+     (reduce conj! res e)
+     (reduce conj! res f)
+     (persistent! res))))
```

Comment:
Should we have an `& more` arity? I don't know what the performance implications are of having one, if it's never called.

Reply:
~There are no implications, but the lack of `& more` is intentional. If anybody needs to concat more sequences, they are welcome to add another arity instead of silently using a less efficient varargs version.~

Actually, the cost of lumping lists together will certainly dominate any overhead that comes from one measly varargs wrap. I'll redo.